File: Algorithms/1299. Replace Elements with Greatest Element on Right Side/golang/solution.go
func replaceElements(arr []int) []int {
	mx := -1
	for i := len(arr) - 1; i >= 0; i-- {
		if arr[i] > mx {
			arr[i], mx = mx, arr[i]
		} else {
			arr[i] = mx
		}
	}
	return arr
}

File: Algorithms/1299. Replace Elements with Greatest Element on Right Side/swift/solution.swift
class Solution {
    func replaceElements(_ arr: [Int]) -> [Int] {
        var mx: Int = -1
        var res: [Int] = arr
        for i in (0..<arr.count).reversed() {
            res[i] = mx
            mx = max(mx, arr[i])
        }
        return res
    }
}